src/sinks/kafka.rs

```diff
@@ -405,7 +420,7 @@ fn encode_event(
     mut event: Event,
     key_field: &Option<String>,
     encoding: &EncodingConfig<Encoding>,
-) -> (Vec<u8>, Vec<u8>) {
```

Comment:
It'd be nice if `encode_event` didn't need to worry about the metadata. Would it make sense to have something like the `into_parts` method to split out the metadata before encoding? Or even something like `take_metadata`?

I realize it's probably somewhat complicated here by working on both logs and metrics, but ideally that wouldn't matter.

Reply:
So, long story here. When I was first developing the initial dummy event metadata, I started with turning `Event` into a new wrapper struct containing the metadata and the actual event enum. This ended up requiring major code changes all over the place, and numerous conversion traits, etc. So, I reworked it to store the event metadata alongside the log and metric data, which required only a relatively few changes.

This unfortunately means that a `fn Event::into_parts` is simply not possible, as `Event` is still an enum with variable parts behind it. `fn Event::take_metadata` is a possibility, calling down to an equivalent function in each variant, which in turn do a `swap` with `EventMetadata::default()`, which doesn't currently allocate any memory or do other real work.

As this goes beyond just the `kafka` sink, I'll open up a separate issue to track this.